### PR TITLE
:bug: Fixes after the rebase on the Syncer refactoring

### PR DESCRIPTION
**Pull request by davidfestal**
Signed-off-by: David Festal <dfestal@redhat.com>

## Summary

Fixes after the rebase on the Syncer refactoring

## Disclaimer

Still to be done after this PR is merged to https://github.com/kcp-dev/kcp/pull/2214 :
- Cleanup the logic that creates / updates the resource in upstream: updating the `spec` sub-resource doesn't make sense.
- Rebase on top of main

### CODE REVIEW

File: pkg/syncer/upsync/upsync_controller.go [hunk 1]
```diff
@@ -199,46 +193,60 @@ func NewUpSyncer(syncerLogger logr.Logger, syncTargetWorkspace logicalcluster.Na
...
+			informer, ok := informers[gvr]
+			if !ok {
+				if shared.ContainsGVR(notSynced, gvr) {
+					return nil, fmt.Errorf("informer for gvr %v not synced in the downstream informer factory", gvr)
+				}
+				return nil, fmt.Errorf("gvr %v should be known in the downstream informer factory", gvr)
+			}
+			return informer.Lister(), nil
+		},
+		syncTargetName:      syncTargetKey,
```
Comment (yana1205) [inline pkg/syncer/upsync/upsync_controller.go:207]: Hi, excuse me for interrupting. 
I am very interesting to Upsync feature and am trying it on this branch.
Maybe, is this typo?
`syncTargetName:      syncTargetKey,` => `syncTargetName:      syncTargetName,`?
I'm facing an issue at https://github.com/davidfestal/kcp/blob/pr/bipuladh/2214/pkg/syncer/upsync/upsync_process.go#L60 that calculates the downstream namespace hash prefix but results in a different value from the actually used one.

**Comment by davidfestal** [comment]
Thanks @yana1205. I'll have a look into it.
I'm still polishing the fixes on the initial PR, and will update this PR when finished.

**Comment by davidfestal** [comment]
Yes, its seems a bug. I'll include the fix in my current changes

**Comment by yana1205** [comment]
Thanks! @davidfestal

**Pull request merged**